clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -991,10 +990,13 @@ void SYCLToolChain::AddImpliedTargetArgs(const llvm::Triple &Triple,
       CmdArgs.push_back("-device");
       CmdArgs.push_back(Args.MakeArgString(DepInfo));
     }
-    // -ftarget-compile-fast
+    // -ftarget-compile-fast AOT
     if (Args.hasArg(options::OPT_ftarget_compile_fast)) {
       BeArgs.push_back("-igc_opts 'PartitionUnit=1,SubroutineThreshold=50000'");
     }
+  } else if (!Triple.isNVPTX() && !Triple.isAMDGCN()) {
```

Comment:
The documentation says "Currently only supported on Intel GPUs.", but I read this as "Currently everything except NVPTX and AMDGDN targets are supported".
Is it possible to replace this check with Intel GPU (i.e. SPIR-V) target?

Reply:
@sarnex, how does adding `Triple.getSubArch() == llvm::Triple::NoSubArch` resolve the comment above?

Reply:
@sarnex, ping.

Reply:
> @bader this is JIT, so we don't know what the hardware is in the driver, we can't check for intel gpu jit here. the NVPTX and AMDGCN checks are to prevent us to go in this if for AOT with those targets. intel gpu aot (gen) is checked above, and the other jit spir subtargets are fpga and cpu, so checking nosubarch makes sure we don't go in for those two cases. did i misunderstand your comment?

I think we better check for one supported target. Ruling out unsupported targets doesn't scale.

Reply:
good point, thanks, will do